src/boi/boi_wifi.cpp

```diff
@@ -105,162 +88,131 @@ void boi_wifi::DisableWiFi()
+            }
         }
-
         pthread_mutex_unlock(&lock);
-
         yield();
-        delay(100);
+        delay(100); //read and send sensor data every X milliseconds
     };
-
     pthread_mutex_unlock(&lockDone);
 }
 
-void boi_wifi::Reconfigure(const OptionsStruct *Options)
-{
+void boi_wifi::Reconfigure(const OptionsStruct *Options) {
     char WifiName[20 + 1 + 10];
     // uint8_t BatterySymbol[] = " \xF0\x9F\x94\x8B "; // Standard edition, but who wants that?
     // uint8_t BatterySymbol[] = " \xF0\x9F\x8E\x83 "; // SPOOKY EDITION OOOOOOOOOoOOOoOooooooooo.............
     uint8_t BatterySymbol[] = " \xF0\x9F\x8E\x83 \xF0\x9F\x98\xB7 \xF0\x9F\x8E\x83 "; // SAFE MODE WITH NETWORKING EDITION - WASH YOUR HANDS!!!!!
-
     memcpy(WifiName, &BatterySymbol[1], 5);
 
-    //reconfigure the wifi with batteries around it
+        //reconfigure the wifi with batteries around it
     memcpy(&WifiName[5], Options->WifiName, strlen(Options->WifiName));
     memcpy(&WifiName[5 + strlen(Options->WifiName)], BatterySymbol, 5);
     WifiName[10 + strlen(Options->WifiName)] = 0;
     Serial.printf("Wifi portal name: %s\n", WifiName);
 
-    if(!WiFi.softAP(WifiName, Options->WifiPassword))
+    if(!WiFi.softAP(WifiName, Options->WifiPassword)) {
         Serial.print("Error calling WiFi.softAP\n");
+    }
 }
 
-void boi_wifi::setup_captive_portal(const OptionsStruct *Options){
-    const byte        DNS_PORT = 53;          // Capture DNS requests on port 53
+void boi_wifi::setup_captive_portal(const OptionsStruct *Options) {
+    const byte DNS_PORT = 53; // Capture DNS requests on port 53
     // TODO: Update to internal ip address
-    // WiFi.localIP();
     IPAddress ip_address = WiFi.localIP(); //captive portal uses self ip
 
     //IPAddress apIP(ip_address[0], ip_address[1], ip_address[2], ip_address[3]); // Private network for server
     IPAddress apIP(1,1,1,1); // Private network for server
 
-    if(this->ServerCheckThread)
-    {
+    if(this->ServerCheckThread) {
         this->DisableWiFi();
         yield();
         delay(1000);
     }
     
-    if(!WiFi.mode(WIFI_AP_STA))
+    if(!WiFi.mode(WIFI_AP_STA)) {
         Serial.print("Error setting WiFi mode\n");
+    }
 
     this->Reconfigure(Options);
 
-    if(!WiFi.softAPConfig(apIP, apIP, IPAddress(255, 255, 255, 0)))
+    if(!WiFi.softAPConfig(apIP, apIP, IPAddress(255, 255, 255, 0))) {
         Serial.print("Error calling WiFi.softAPConfig\n");
+    }
 
-    // if DNSServer is started with "*" for domain name, it will reply with
-    // provided IP to all DNS request
     this->dnsServer = new DNSServer();
-    this->dnsServer->start(DNS_PORT, "*", apIP);
+    this->dnsServer->start(DNS_PORT, "*", apIP); // if DNSServer is started with "*" for domain name, it will reply with provided IP to all DNS request
 
-    // reply to all requests with same HTML
+    // reply to all requests with same HTML - TODO is this happening somewhere, or should this be deleted?
```

Comment:
This was removed as it was crashing iphones. We only respond to requests that are for the IoB, otherwise we reply with an error. Previously, we were causing iPhones to lock up as they were not happy with us just sending back a 200 OK for EVERY REST call made.

Reply:
Actually...if we can reproduce this issue on a modern iphone, we might be able to get a bug bounty from apple...